Example with image rendering functionality

jim-mckeown opened the pull request:
Example from modified fingerprint_simpletest.py adding image viewing and saving functionality using matplotlib and numpy. This has been tested on a Linux PC but should work under Windows or MacOS.

tekktrik commented:
Looks like this is failing the GitHub Actions CI.  You can use `pre-commit` to help find and fix the issues it's having.  You can find information on installing and using `pre-commit` here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code

jim-mckeown commented:
I changed the import order and reformatted with black. Also added non-blocking mode in matplotlib.pyplot so the code will run properly from the command line.

tekktrik commented on examples/fingerprint_picturetest.py:
```diff
@@ -0,0 +1,254 @@
+# SPDX-FileCopyrightText: 2021 ladyada for Adafruit Industries
```
I would change this to your name, as well as remove the credit to Adafruit Industries:

```suggestion
# SPDX-FileCopyrightText: 2023 Jim McKeown
```

tekktrik commented on examples/fingerprint_picturetest.py:
```diff
@@ -0,0 +1,254 @@
+# SPDX-FileCopyrightText: 2021 ladyada for Adafruit Industries
+# SPDX-License-Identifier: MIT
+
+# Added 'View Print' and 'Preview and Find Print' functions to
+# example code fingerprint_simpletest.py
```
We typically use triple quotes for docstrings, but I don't think this is a gating change.  More importantly, I would specify that this example only works on single board computers with the use of Blinka, our CircuitPython compatibility layer.

tekktrik commented on examples/fingerprint_picturetest.py:
```diff
@@ -0,0 +1,254 @@
+# import board
+# import busio
+# from digitalio import DigitalInOut, Direction
+import numpy as np
+from matplotlib import pyplot as plt
+import serial
```
Since these are used in examples, we need to add these as dependencies to `optional_requirements.txt`!

tekktrik reviewed (changes requested):
Thanks for the submission!  I added some feedback, let me know if I can expand on anything.  I can see if anyone can test this too, which would be good if we can arrange.

jim-mckeown commented:
Please have a look at the latest changes and let me know if any others are needed or would be helpful. Thanks.

tekktrik commented on optional_requirements.txt:
```diff
@@ -1,3 +1,7 @@
+
+numpy
+matplotlib
+serial
```
This should be `pyserial`:

```suggestion
pyserial
```

tekktrik commented on examples/fingerprint_picturetest.py:
```diff
@@ -0,0 +1,250 @@
+import serial
+import adafruit_fingerprint
+
+"""Added 'View Print' and 'Preview and Find Print' to fingerprint_simpletest.py"""
+"""This example only works on single board computers with the use of Blinka"""
```
Rather than note the difference from `fingerprint_simpletest.py`, let's ouright say what this example will do so users will immediately know :)

Also, I would move this above the import statements and use only a single set of triple quotes, like this:

```python
# SPDX-FileCopyrightText: 2023 Jim McKeown
# SPDX-License-Identifier: MIT

"""
This is the module docstring!
"""

import time
...
```

tekktrik reviewed (changes requested):
Thanks for the changes!  Just a few small tweaks, but we'll see if I or someone else can test this after these get updated!

jim-mckeown commented:
Please have a look at the latest changes.

Just one thought about pyserial. Since it is already in requirements.txt, does it also need to be in optional_requirements.txt?

FoamyGuy reviewed (approved):
Looks good to me. 

All requested changes have been addressed with newer commits.
